Document view

markuswich opened the pull request:
Added sortable document table to the home screen

alexanderbarnhill commented on documentdata.py:
```diff
@@ -0,0 +1,19 @@
+        self.size = size
+        self.createdAt = createdAt
+
+    def asDict(self):
+        return {
+            "name" : self.name,
+            "path" : self.path,
+            "type" : self.type,
+            "lang" : self.lang,
+            "size" : self.size,
+            "createdAt" : self.createdAt
```
~~Is this necessary? I would have thought that returning a jsonified object would return exactly this?~~

It is necessary. My mistake

alexanderbarnhill commented on documentdata.py:
```diff
@@ -0,0 +1,19 @@
+        self.size = size
+        self.createdAt = createdAt
+
+    def asDict(self):
```
```suggestion
    def as_dict(self):
```
No camel casing in python :)

markuswich commented:
-Fixed the naming.
-I renamed apiUrl back to serverUrl to be in line with Nazmus' code
-Fixed a bug where the timestamp wasn't correctly send to client and thus broke sorting; timestamps also support hours/minutes/seconds now

Do you think i should add runtime type checking for the constructor of DocumentData?

alexanderbarnhill commented:
> -Fixed the naming.
> -I renamed apiUrl back to serverUrl to be in line with Nazmus' code
> -Fixed a bug where the timestamp wasn't correctly send to client and thus broke sorting; timestamps also support hours/minutes/seconds now
> 
> Do you think i should add runtime type checking for the constructor of DocumentData?

It wouldn't hurt, but we can also do that later if we find it necessary

markuswich commented:
I just noticed I should probably add comments to the python code. Do we have a style for that or are we going to use something like doxygen?

alexanderbarnhill commented:
> I just noticed I should probably add comments to the python code. Do we have a style for that or are we going to use something like doxygen?

I've just been going with the standard pydoc format. But its still early enough that we can change that if you want.
